docker/update_or_build.sh

```diff
@@ -0,0 +1,99 @@
+#!/bin/bash
+# Copyright (c) The Libra Core Contributors
+# SPDX-License-Identifier: Apache-2.0
+
+function usage {
+  echo "Usage:"
+  echo "build and properly tags images for deployment to docker hub"
+  echo "update_or_build.sh [-p] -g <GITHASH> -b <TARGET_BRANCH> -n <image name> [-u]"
+  echo "-p indicates this a prebuild, where images are built and pushed to dockerhub with an infix of '_pre_', should be run on the 'auto' branch, trigger by bors."
+  echo "-b the branch we're building on, or the branch we're targeting if a prebuild"
+  echo "-n name, one of init, mint, validator, validator-dynamic, safety-rules, cluster-test"
+  echo "-u 'upload', or 'push' the docker images will be pushed to dockerhub, otherwise only locally tag"
+  echo "should be called from the root folder of the libra project, and must have it's .git history"
+}
+
+PREBUILD=false;
+NAME=
+BRANCH=
+PUSH=false
+
+#parse args
+while getopts "pb:n:u" arg; do
+  case $arg in
+    p)
+      PREBUILD="true"
+      ;;
+    n)
+      NAME=$OPTARG
+      ;;
+    b)
+      BRANCH=$OPTARG
+      ;;
+    u)
+      PUSH="true"
+      ;;
+    h)
+      usage;
+      exit 0;
+      ;;
+  esac
+done
+
+GIT_REV=$(git rev-parse --short=8 HEAD)
+
+[ "$BRANCH" != "" ] || { echo "-b branch must be set"; usage; exit 99; }
+[ "$GIT_REV" != "" ] || { echo "Could not determine git revision, aborting"; usage; exit 99; }
+[ "$NAME" != "" ] || { echo "-n name must be set"; usage; exit 99; }
+
+PULLED="-1"
+
+#Convert dashes to underscores to get tag names, except for validator, which for some reason is "e2e"
+tag_name=`echo $NAME | sed 's/-/_/g'`
+if [ $NAME == "validator" ]; then
+  tag_name="e2e";
+fi
```

Comment:
It looks like we should clean up https://github.com/libra/libra/blob/master/docker/validator-dynamic/Dockerfile instead such that we don't muck with the intermediate layer.

Reply:
Agreed, but not it this pr.   Probably plenty of terraform nasties out there.   A quick gh search found a lot of results.